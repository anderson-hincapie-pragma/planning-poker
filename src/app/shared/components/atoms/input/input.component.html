<input class="a-input" [type]="type" [placeholder]="placeholder" [formControl]="control" />