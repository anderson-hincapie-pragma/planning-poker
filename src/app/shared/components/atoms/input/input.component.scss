:host {
    display: contents;
}
.a-input {
    padding: 4px 4px 4px 15px;
    width: 100%;
    height: 40px;
    background: rgba(100, 41, 205, 0.08);
    border: 1px solid #BB65FF;
    border-radius: 100px;
    font-family: 'Poppins';
    font-weight: 400;
    font-size: 14px;
    line-height: 140%;
    color: #FFFFFF;
}