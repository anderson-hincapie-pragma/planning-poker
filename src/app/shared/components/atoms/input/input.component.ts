import { Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'app-input',
  templateUrl: './input.component.html',
  styleUrls: ['./input.component.scss']
})
export class InputComponent {

  @Input()
  public type: 'email' | 'number' | 'search' | 'text' = 'text';

  @Input()
  public placeholder: string = '';

  @Input()
  public control: FormControl = new FormControl();

}
