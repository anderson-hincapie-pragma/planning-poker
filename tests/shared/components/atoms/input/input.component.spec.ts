import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { InputComponent } from '@shared/components/atoms/input/input.component';

describe('InputComponent', () => {
  let component: InputComponent;
  let fixture: ComponentFixture<InputComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule],
      declarations: [InputComponent]
    });
    fixture = TestBed.createComponent(InputComponent);
    compiled = fixture.nativeElement;
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('shoud match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

});
