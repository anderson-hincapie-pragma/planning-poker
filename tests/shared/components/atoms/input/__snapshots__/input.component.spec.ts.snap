// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent shoud match snapshot 1`] = `
<div
  id="root1"
>
  <input
    class="a-input ng-untouched ng-pristine ng-valid"
    placeholder=""
    type="text"
  />
</div>
`;
